test: rename "cql-pytest" to "cqlpy"

nyh opened the pull request:
Python and Python developers don't like directory names to include a  minus sign, like "cql-pytest". In this patch we rename test/cql-pytest to test/cqlpy, and also change a few references in other code (e.g., code that used test/cql-pytest/run.py) and also references to this test suite in documentation and comments.
    
Arguably, the word "test" was always redundant in test/cql-pytest, and I want to leave the "py" in test/cqlpy to emphasize that it's Python-based tests, contrasting with test/cql which are CQL-request-only approval tests.

The second patch in the series fixes a small regression in the test/cqlpy/run script.
    
Fixes #20846

Test organization only, no backports needed.

scylladb-promoter commented:
## Docs Preview :book:
Docs Preview for this pull request is available [here](http://previews.docs.scylladb.com/scylladb/21446)

**Changed Files:**
- [cql/wasm](http://previews.docs.scylladb.com/scylladb/21446/cql/wasm)

**Note:** This preview will be available for 30 days and will be automatically deleted after that period. You can manually trigger a new build by committing changes.

xtrey commented:
Firstly, I suppose we agreed to rename cql to cql_approval and cql-pytest to cql. Secondly, I think backport is needed to all active versions, since the next change in the new directory will not be possible to backport anymore. And finally, almost identical [commit](https://github.com/xtrey/scylladb/commit/2d57e38cf48990bb044a621655990b4e6c0a06c9) already exists, so why do we need this one?

nyh commented:
> Firstly, I suppose we agreed to rename cql to cql_approval and cql-pytest to cql. 

This alternative was raised once, but I don't remember anyone agreed to it. We have issue https://github.com/scylladb/scylladb/issues/20846 and nobody commented that we should do something else than what I wrote there.

Anyway, I won't complain too loudly if my favorite test suite (test/cql-pytest) will get the shorter name "cql" and the existing suite with that name by @kostja  will get the ugly name cql_approval, but I'm not sure how much @kostja will appreciate this.

Finally, renaming to an existing name (cql) can create confusion in backports - consider a patch that introduces a new file "test/cql/something.py". This patch will seem to be successfully merged on an old branch that had test/cql for approval tests - but after the merge, this test will be ignored (because it doesn't have the name "*.cql").

> Secondly, I think backport is needed to all active versions, since the next change in the new directory will not be possible to backport anymore.

This is true, but I don't know if we really want to backport this sort of massive change. We'll need to discuss this further...

> And finally, almost identical [commit](https://github.com/xtrey/scylladb/commit/2d57e38cf48990bb044a621655990b4e6c0a06c9) already exists, so why do we need this one?

What do you mean this commit "exists"? What pull request is it part of? I didn't see one...

In any case, your commit modifies hundreds of files to replace things like
```
from cassandra_tests.porting import *
```
by
```
from test.cqlpy.cassandra_tests.porting import *
```
there is no reason why these extra of changes need to be done with, or discussed with, the changes in this patch.
Yes, the name "cqlpy" is now more friendly to Python. We can use this friendliness now. But we need to do it in separate patches that each carries a compelling explanation why it's a good idea, and why it doesn't break existing use cases (e.g., `cd test/cqlpy; pytest` or `test/cqlpy/run-cassandra`.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12985/)
:x: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12985/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: README.md
:small_blue_diamond: alternator
:small_blue_diamond: alternator/test_cql_rbac
:small_blue_diamond: alternator/test_table
:small_blue_diamond: boost
:small_blue_diamond: redis
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api/test_column_family
:small_blue_diamond: rest_api/test_compaction_manager
:small_blue_diamond: rest_api/test_compaction_task
:small_blue_diamond: rest_api/test_compactionhistory
:small_blue_diamond: rest_api/test_gossiper
:small_blue_diamond: rest_api/test_repair_task
:small_blue_diamond: rest_api/test_storage_service
:small_blue_diamond: rest_api/test_task_manager
:small_blue_diamond: topology/test_mv
:small_blue_diamond: topology_custom/test_multidc
:small_blue_diamond: topology_custom/test_tablets
#### Failed Tests (1/16963):
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/12985/testReport/junit/boost.secondary_index_test.test_secondary_index_case_compound_partition_key.1.xunit/xml/_empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 3 hr 25 min
- Builder: i-0bb59b95c05f3193a (m5ad.8xlarge)

nyh commented:
CI failed yet again on https://github.com/scylladb/scylladb/issues/20772. Restarting it.

kbr-scylla commented:
> This is true, but I don't know if we really want to backport this sort of massive change. We'll need to discuss this further...

I think we can backport without much trouble. Git is pretty smart about renames. Not backporting the change will inflict more pain long-term

kbr-scylla commented:
LGTM, but `next` is locked. Will queue once unlocked

> CI failed yet again on https://github.com/scylladb/scylladb/issues/20772. Restarting it.

I'd say there's no need, we're just burning $.

nyh commented:
> I think we can backport without much trouble. Git is pretty smart about renames. Not backporting the change will inflict more pain long-term

Ok, you guys convinced me. Added the backports tags instead of no-backports.

nyh commented:
> > What do you mean this commit "exists"? What pull request is it part of? I didn't see one...
> 
> In your issue, there is an activity where stated that one commit mentioned it in the same way as your current PR connected to the #20846

I saw this commit "mentioned" today, I didn't know what it means, it's not part of any pull request you ever sent. When I looked at it I saw you mixed the rename with other changes like changing all the imports and didn't want to do all of that at once.
 
> > Anyway, I won't complain too loudly if my favorite test suite (test/cql-pytest) will get the shorter name "cql" and the existing suite with that name by @kostja will get the ugly name cql_approval, but I'm not sure how much @kostja will appreciate this.
> 
> From my understanding, it was agreed in the email thread where you were involved. I suppose Kostja has no objection to the naming.

Searching my mail for `cql_approval` I indeed found a mail by @kostja where he said
```
It can be moved to "approval" or "cql_approval" or "approval_tests". 
```
but also:
```
OK, I think cqlpy is good enough.
```

So I never felt there was any "decision"...
By the way, if you understood the decision was to use the name "cql", why does your commit (which you mention above) also renamed it to cqlpy?

> > Finally, renaming to an existing name (cql) can create confusion in backports - consider a patch that introduces a new file "test/cql/something.py". This patch will seem to be successfully merged on an old branch that had test/cql for approval tests - but after the merge, this test will be ignored (because it doesn't have the name "*.cql").
> 
> This is precisely why we need to backport to the active branches.

Well, almost... Sometimes we also do backports to various inactive or secret branches. At that point you can have a backport that seems to be successful (create a file in test/cql) but doesn't do what you thought it does.
I agree it's not a huge problem. If we are convinced we should rename this directory test/cql, this rare backporting reason should be what holds us back.

> 
> > there is no reason why these extra of changes need to be done with, or discussed with, the changes in this patch.
> 
> With only current changes, your tests will not be executed with test.py. So you just switched them off for CI.

Wait, what? Why? We have test/cqlpy/suite.yaml just like we previously had test/cql-pytest/suite.yaml. Why won't test.py notice that? What am I missing?

>  Try to find the results of the test_bloom_filter in your execution and compare it to this [one](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12973/)

```
$ ./test.py --mode=dev test_bloom_filter
Found 1 tests.
================================================================================
[N/TOTAL]   SUITE    MODE   RESULT   TEST
------------------------------------------------------------------------------
[1/1]       cqlpy     dev   [ PASS ] cqlpy.test_bloom_filter.1
------------------------------------------------------------------------------
CPU utilization: 8.2%
```

What's wrong with that?

> > Yes, the name "cqlpy" is now more friendly to Python. We can use this friendliness now. But we need to do it in separate patches that each carries a compelling explanation why it's a good idea, and why it doesn't break existing use cases (e.g., cd test/cqlpy; pytest or test/cqlpy/run-cassandra.
> 
> I made the same mistake several times. I've changed test.py and checked if it's working with it without checking the pytest. So seems you to made it as well. Did you executed cqlpy directory through test.py?

No, I was expecting the CI to do this for me. But now that I did run it manually (see example above), it does seem to run. Why do you think it wouldn't run because?

I also tried running all the tests:
```
$ ./test.py --mode=dev cqlpy            
Found 121 tests.
...
```

xtrey commented:
> Wait, what? Why? We have test/cqlpy/suite.yaml just like we previously had test/cql-pytest/suite.yaml. Why won't test.py notice that? What am I missing?

Oh, sorry, my bad. You renamed without making it a python module, that's why it's working. The main goal was to make this directory as a module, renaming is just a road block on the way. I was confused why it's working at all. 
So, it's working as previously. But it's weird why it's not shown in the result. Anyway, my assumption was incorrect.

> By the way, if you understood the decision was to use the name "cql", why does your commit (which you mention above) also renamed it to cqlpy?

Because I did this, and then started that email thread. I decided that changing name pretty easy when all imports already in place.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12999/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12999/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: README.md
:small_blue_diamond: alternator
:small_blue_diamond: alternator/test_cql_rbac
:small_blue_diamond: alternator/test_table
:small_blue_diamond: boost
:small_blue_diamond: redis
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api/test_column_family
:small_blue_diamond: rest_api/test_compaction_manager
:small_blue_diamond: rest_api/test_compaction_task
:small_blue_diamond: rest_api/test_compactionhistory
:small_blue_diamond: rest_api/test_gossiper
:small_blue_diamond: rest_api/test_repair_task
:small_blue_diamond: rest_api/test_storage_service
:small_blue_diamond: rest_api/test_task_manager
:small_blue_diamond: topology/test_mv
:small_blue_diamond: topology_custom/test_multidc
:small_blue_diamond: topology_custom/test_tablets
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/108/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/1562/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/12663/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/4636/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12999/)
#### Build Details:
- Duration: 10 hr
- Builder: i-070939e1c6ae34621 (m5ad.12xlarge)

xemul commented:
Conflicts, please rebase

kbr-scylla commented:
We can do the rename in this PR, and change it into a proper module in a follow-up.

nyh commented:
rebased to eliminate conflicts

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/13019/)
:x: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/13019/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: README.md
:small_blue_diamond: alternator
:small_blue_diamond: alternator/test_cql_rbac
:small_blue_diamond: alternator/test_table
:small_blue_diamond: boost
:small_blue_diamond: redis
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api
:small_blue_diamond: rest_api/test_column_family
:small_blue_diamond: rest_api/test_compaction_manager
:small_blue_diamond: rest_api/test_compaction_task
:small_blue_diamond: rest_api/test_compactionhistory
:small_blue_diamond: rest_api/test_gossiper
:small_blue_diamond: rest_api/test_repair_task
:small_blue_diamond: rest_api/test_storage_service
:small_blue_diamond: rest_api/test_task_manager
:small_blue_diamond: topology/test_mv
:small_blue_diamond: topology_custom/test_multidc
:small_blue_diamond: topology_custom/test_tablets
#### Failed Tests (1/39572):
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/13019/testReport/junit/boost.secondary_index_test.test_cannot_drop_secondary_index_backing_mv.1.xunit/xml/_empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 5 hr 22 min
- Builder: i-070c132ad51426426 (m5ad.8xlarge)

nyh commented:
Crashed as usual on a random secondary-index test during shutdown. https://github.com/scylladb/scylladb/issues/20772

kbr-scylla commented:
qd

kbr-scylla commented:
> Yes, but "change it to a proper module" will have to be justified.

For example, running Python static analysis code tools, which like when the code conforms to Python specs, in particular, when they understand the hierarchy of modules

xtrey commented:
> Yes, but "change it to a proper module" will have to be justified.

It's already justified: "One cluster pool for all tests". To achieve this, all tests should be possible to execute with one command: `pytest`. To make this happen, all should be modules.
